tests/micro/qemu/zephyr-runtime/src/main.c

```diff
@@ -161,6 +162,27 @@ tvm_crt_error_t TVMPlatformTimerStop(double* elapsed_time_seconds) {
   return kTvmErrorNoError;
 }
 
+tvm_crt_error_t TVMPlatformGenerateRandom(uint8_t* buffer, size_t num_bytes) {
+  uint32_t random;  // one unit of random data.
```

Comment:
Maybe use modulo operator and remove `num_full_blocks`? Like:

```suggestion
  uint32_t random;  // one unit of random data.

  size_t block_size = sizeof(random);
  size_t num_blocks = num_bytes / block_size;
  size_t num_tail_bytes  = num_bytes % block_size;

  // Fill parts of `buffer` which are as large as `random`.
  for (int i = 0; i < num_blocks; ++i) {
    random = sys_rand32_get();
    memcpy(&buffer[i * block_size], &random, block_size);
  }

  // Fill any leftover tail which is smaller than `random`.
  if (num_tail_bytes > 0) {
    random = sys_rand32_get();
    memcpy(&buffer[num_bytes - num_tail_bytes], &random, num_tail_bytes);
  }
```

Reply:
simplified the tail block computation at the end of mine using modulo operator. thanks for the suggestion!